lib/logstash/filters/grok/timeout_enforcer.rb

```diff
@@ -19,19 +19,15 @@ def grok_till_timeout(grok, field, value)
       grok.execute(value)
     rescue InterruptedRegexpError => e
       raise ::LogStash::Filters::Grok::TimeoutException.new(grok, field, value)
+    rescue java.lang.InterruptedException => e
+      # NOOP, we don't expect these, but maybe some interruptible thing could be
+      # added to grok besides regexps
+      @logger.debug("Unexpected interruptible caught during grok. This isn't a problem most likely")
     ensure
-      unless stop_thread_groking(thread)
-        @cancel_mutex.lock
-        begin
-          # Clear any interrupts from any previous invocations that were not caught by Joni
-          # It may appear that this should go in #stop_thread_groking but that would actually
-          # break functionality! If this were moved there we would clear the interrupt
-          # immediately after setting it in #cancel_timed_out, hence this MUST be here
-          java.lang.Thread.interrupted
-        ensure
-          @cancel_mutex.unlock
-        end
-      end
+      # If the regexp finished, but interrupt was called after, we'll want to
+      # clear the interrupted status anyway
+      @threads_to_start_time.remove(thread)
```

Comment:
I've opted to remove `start_thread_groking` and just inline its one call. `stop_thread_grokking` would only be called once from one spot since there isn't a single way to stop it. In one spot we must `compute` in another `remove`. 

WDYT of how it looks now?

Reply:
eh, in an ideal world we'd have better encapsulation (e.g., both `grok_till_timeout ` and `cancel_time_out!` reach into the `@threads_to_start_time` ivar directly to muck with its internals), but at least this method isn't mixing abstractions.

Reply:
☕️